go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -277,50 +275,58 @@ func (ovn *Controller) deleteLoadBalancerRejectACL(lb, vip string) {
 			klog.Errorf("Unable to parse vip for Reject ACL deletion: %v", err)
 			return
 		}
-		ovn.removeStaleRejectACL(lb, ip, port)
+		aclUUID, err = ovn.removeStaleRejectACL(lb, ip, port)
+		if err == nil {
+			ovn.removeACLFromPortGroup(lb, aclUUID)
+		}
+		return
 	} else if aclUUID == "" {
 		// Must have endpoints and no reject ACL to remove
 		klog.V(5).Infof("No reject ACL found to remove for load balancer: %s, vip: %s", lb, vip)
 		return
-	} else {
-		ovn.removeACLFromNodeSwitches(lb, aclUUID)
 	}
+	ovn.removeACLFromPortGroup(lb, aclUUID)
 	ovn.removeServiceACL(lb, vip)
 }
 
-func (ovn *Controller) removeStaleRejectACL(lb, ip string, port int32) {
+func (ovn *Controller) removeStaleRejectACL(lb, ip string, port int32) (string, error) {
```

Comment:
this function no longer removes a stale reject ACL, it just finds an acl. Please rename it